### PR TITLE
Kill Mechanics for Infiltrators (#193)

### LINKED ISSUE #193: Kill mechanics for infiltrators
A kill mechanic for players assigned as an "infiltrator". Used to kill others not assigned infiltrator.

Should have:
- Reloading (triggered by player, has a cool-down time)
- Adjustable range
- A shortcut, and a GUI icon
- Toggle-able based on other events

See https://github.com/opensuspect/design-document/blob/main/general_gameplay.md.

### PR DESCRIPTION
**Pull request by jngo102**
- Added red outline to players when they are in kill range of an infiltrator
- Added kill icon that indicates cooldown between kills
- Added kill and reload input actions for infiltrators
- Check role to determine whether player should have infiltrator elements
- Renamed and changed some physics layer names (tentative names)

### CODE REVIEW

**Review by TheSecondReal0 — CHANGES_REQUESTED** [review]
Other players are visible even when out of range of the light circle, this was probably caused by the shader on the player sprite being changed. Besides this, it looks great!

**Comment by Poket-Jony** [comment]
I also noticed that the infiltrator-ui does not reset itself when you go back to the lobby. In addition, the kill countdown doesn't work for me.

**Comment by jngo102** [comment]
Can you clarify what you mean by the cooldown not working?

EDIT: The former issue may have been fixed in the most recent commit.

File: src/assets/player/infiltrator.gd [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+# Scene containing UI element for kill indicator
+onready var killui_scene: PackedScene = load("res://assets/ui/hud/infiltratorui/killui.tscn")
+
+# UI Controller for the player
+onready var ui_controller: CanvasLayer = get_tree().get_root().find_node("uicontroller", true, false)
+# Area within which a player may be killed
+onready var kill_area: Area2D = $KillArea
+# Cooldown before next kill may be made
+onready var kill_cooldown_timer: Timer = $KillCooldownTimer
```
Comment (nicemicro) [inline src/assets/player/infiltrator.gd:12]: Please check the game design document on [kill mechanics](https://github.com/opensuspect/design-document/blob/main/general_gameplay.md#kill-mechanics).
While a kill cooldown is possible to be used on other maps / game modes, in the first map there is no cooldown, but a gun reloading mechanism, that is initiated by the player, but while reloading can't move and can be discovered by others.

**Review by nicemicro — CHANGES_REQUESTED** [review]
Please check [kill mechanics](https://github.com/opensuspect/design-document/blob/main/general_gameplay.md#kill-mechanics) and conform to the description provided there.

**Review by TheSecondReal0 — CHANGES_REQUESTED** [review]
It works perfectly when it's run in debug mode or in the editor, but the infiltrator client always crashes when the lobby button is pressed. My best idea to fix this is to move away from constantly polling the infiltrator node and instead use signals to relay information.

**Review by nicemicro — DISMISSED** [review]

**Comment by nicemicro** [comment]
Is the mechanics for dying for the killed people will be implemented separately?

**Comment by jngo102** [comment]
> Is the mechanics for dying for the killed people will be implemented separately?

Yes, we should open another PR for that. I'd be willing to take on that task as well.

**Review by TheSecondReal0 — APPROVED** [review]
Everything works perfectly! Despite the long journey to get to this point, it's ready to merge! What a great first PR

**Pull request merged**